File: leetcode/easy/calculate-digit-sum-of-a-string.ts
// https://leetcode.com/problems/calculate-digit-sum-of-a-string/

// MINE
function digitSum(s: string, k: number): string {
  while (s.length > k) {
    const A = []
    for (let i = 0; i < s.length; i += k) {
      A.push(
        s
          .slice(i, Math.min(i + k, s.length))
          .split('')
          .reduce((acc, cur) => acc + parseInt(cur), 0)
      )
    }
    s = A.join('')
  }
  return s
}

// ANSWER

File: leetcode/easy/calculate-digit-sum-of-a-string.py
# https://leetcode.com/problems/calculate-digit-sum-of-a-string/

class Solution:
    def digitSum(self, s: str, k: int) -> str:
        while len(s) > k:
            A = [s[i:i+k] for i in range(0, len(s), k)]
            s = ''
            for ds in A:
                s += str(sum([int(d) for d in ds]))
        return s
